### PR TITLE
new entity type storing HTML reports associated with Tables

### PR DESCRIPTION
**Pull request by dechoma**
Resolve lyft/amundsen#525

### CODE REVIEW

File: amundsenatlastypes/schema/01_4_table_report.json [hunk 1]
```diff
@@ -0,0 +1,57 @@
+{
+  "entityDefs": [
+    {
+      "superTypes": [
+        "DataSet"
+      ],
+      "name": "TableReport",
```
Comment (mgorsk1) [inline amundsenatlastypes/schema/01_4_table_report.json:7]: this should be named `Report`

File: amundsenatlastypes/schema/01_4_table_report.json [hunk 2]
```diff
@@ -0,0 +1,57 @@
+{
+  "entityDefs": [
+    {
+      "superTypes": [
+        "DataSet"
```
Comment (mgorsk1) [inline amundsenatlastypes/schema/01_4_table_report.json:5]: should be `Referenceable`

File: amundsenatlastypes/schema/01_4_table_report.json [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+          "valuesMaxCount": 1,
+          "isUnique": false,
+          "isIndexable": true
+        }
+      ]
+    }
+  ],
+  "relationshipDefs": [
+    {
+      "name": "Table__TableReport",
```
Comment (mgorsk1) [inline amundsenatlastypes/schema/01_4_table_report.json:36]: the relationship should be more generic - between `Report` and `Referenceable`. This way report can be connected to any type of entity.

File: amundsenatlastypes/__init__.py [hunk 4]
```diff
@@ -75,6 +75,9 @@ def create_reader_schema(self):
     def create_bookmark_schema(self):
         self.create_or_update(self.get_schema_dict(bookmark_schema), "Bookmark")
 
+    def create_table_report_schema(self):
```
Comment (mgorsk1) [inline amundsenatlastypes/__init__.py:78]: please rename to create_report_schema

File: amundsenatlastypes/schema/01_4_table_report.json (hunk 1 above)
Comment (dechoma) [reply amundsenatlastypes/schema/01_4_table_report.json]: changed in 030fe2bce628b835c902856811312dc7a37ae6cb

File: amundsenatlastypes/schema/01_4_table_report.json (hunk 2 above)
Comment (dechoma) [reply amundsenatlastypes/schema/01_4_table_report.json]: changed in 030fe2b

File: amundsenatlastypes/schema/01_4_table_report.json (hunk 3 above)
Comment (dechoma) [reply amundsenatlastypes/schema/01_4_table_report.json]: changed in 030fe2b

File: amundsenatlastypes/__init__.py (hunk 4 above)
Comment (dechoma) [reply amundsenatlastypes/__init__.py]: changed in 030fe2b

**Comment by dechoma** [comment]
https://github.com/lyft/amundsen/issues/525

File: amundsenatlastypes/schema/01_4_report.json [hunk 5]
```diff
@@ -0,0 +1,57 @@
+{
+  "entityDefs": [
+    {
+      "superTypes": [
+        "Referenceable"
+      ],
+      "name": "Report",
+      "description": "External data reports ( url's) associated with table",
```
Comment (mgorsk1) [inline amundsenatlastypes/schema/01_4_report.json:8]: doesn't have to be associated with table

Comment (dechoma) [reply amundsenatlastypes/schema/01_4_report.json]: fixed in 984cbc1

File: amundsenatlastypes/schema/01_4_report.json [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+      "endDef1": {
+        "type": "Referenceable",
+        "name": "reports",
+        "isContainer": true,
+        "cardinality": "SET",
+        "isLegacyAttribute": true
+      },
+      "endDef2": {
+        "type": "Report",
+        "name": "reportedEntity",
```
Comment (mgorsk1) [inline amundsenatlastypes/schema/01_4_report.json:49]: let's just name it entity

Comment (dechoma) [reply amundsenatlastypes/schema/01_4_report.json]: ok, done

File: amundsenatlastypes/schema/01_4_report.json [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+  "entityDefs": [
+    {
+      "superTypes": [
+        "Referenceable"
+      ],
+      "name": "Report",
+      "description": "External data report (url)",
+      "typeVersion": "2.0",
+      "attributeDefs": [
+				{
```
Comment (mgorsk1) [inline amundsenatlastypes/schema/01_4_report.json:11]: please reformat this json

Comment (dechoma) [reply amundsenatlastypes/schema/01_4_report.json]: done bf4fa15

**Pull request merged**